datasimulator/utils.py

```diff
@@ -1,11 +1,17 @@
-import os
 import random
 import json
 from functools import reduce
 
 from .errors import UserError
 
 
+def attempt(action):
```

Comment:
This appears to be only used in tests, can you move it to /tests and create a test_utils.py or something

Reply:
I think it's designed in a manner to be generally usable anywhere, not exclusively to a test_utils file. Would it be better to make a dedicated utils folder? Otherwise my assumption is that utils is to be used for any general purpose functions.

Reply:
That's true - I guess I don't see the utility or need for this function generally outside of a convenience function for testing whether a given action works. 

I'd much rather manage the exception cases myself with try/except than assigning a boolean variable and doing an if/else. Its a standard way of handling errors and not a new "micro API" (for lack of a better term) that I'd have to learn for thsi repo. Additionally, I suspect most of the code in this repository follows try/except so we should stick with that instead of introducing this to the general codebase. 

Thinking further, while this does help with testing, you could get rid of it entirely and use https://docs.pytest.org/en/stable/reference/reference.html#pytest.raises to verify the exception is thrown as expected rather than checking the true/false of "success"

imo less functions and less code and more using the tools we have.

Reply:
+1 to this, this is what I was getting at here: https://github.com/uc-cdis/data-simulator/pull/105#discussion_r1543454674

Reply:
I think those are all reasonable points. The value proposition is in abstracting error handling to a box to handle at a later time, which has its own uses. In cases where you need to handle the error immediately to do some additional work, that's where original t-c makes perfect sense. 

In this case, we can just use pytest.raises since we only care about catching an exception that occurs. As such the code has been removed since we don't use it here.